app/src/features/apiClient/APIClientManager/workerScripts.ts

```diff
@@ -0,0 +1,138 @@
+/* eslint-disable no-new-func */
+
+import { EnvironmentVariables } from "backend/environment/types";
+
+export const requestWorkerFunction = function (e: MessageEvent) {
+  const { script, request, currentVariables } = e.data;
+
+  const variableSaverQueue: EnvironmentVariables = {};
+  const variableRemoverQueue: string[] = [];
+
+  const sandbox = {
+    rq: {
+      request,
+      environment: {
+        set: (key: string, value: any) => {
+          variableSaverQueue[key] = {
+            localValue: value,
+            syncValue: value,
+            type: typeof value,
+          };
+        },
+        get: (key: string) => {
+          const variable = currentVariables[key];
+          return variable?.localValue || variable?.syncValue;
+        },
+        remove: (key: string) => {
+          variableRemoverQueue.push(key);
+        },
+      },
+    },
+  };
+
+  try {
+    const scriptFunction = new Function(
+      "rq",
+      `
+      "use strict";
+      return (async () => {
+      try {
+        console.log('Executing script');
+        ${script}
+      } catch (error) {
+        console.error('Script error:', error);
+        throw error;
+      }
+      })();
+      `
+    );
+
+    scriptFunction(sandbox.rq).then(() => {
+      this.postMessage({
+        type: "HANDLE_ENVIRONMENT_CHANGES",
+        payload: {
+          currentVariables,
```

Comment:
I don't think we should trust currentVariables from here. We should only return the updated only from here and let the environmentManager merging logic handle non updated variables

Reply:
We do not have merging logic in environmentManager. setVariables always takes in the dump of all the variables